docker/images/interactive/dotnet-spark/Dockerfile

```diff
@@ -4,28 +4,31 @@ FROM dotnet-spark-base-interactive:$DOTNET_SPARK_VERSION
 ARG SPARK_VERSION=3.0.1
 ARG DOTNET_SPARK_JAR="microsoft-spark-3-0_2.12-$DOTNET_SPARK_VERSION"
 
-ENV DAEMON_RUN=true
-ENV SPARK_VERSION=$SPARK_VERSION
-ENV SPARK_HOME=/spark
-
-ENV HADOOP_VERSION=2.7
-ENV PATH="${SPARK_HOME}/bin:${DOTNET_WORKER_DIR}:${PATH}"
-ENV DOTNETBACKEND_PORT=5567
-ENV JUPYTER_ENABLE_LAB=true
+ENV DAEMON_RUN=true \
+    DOTNETBACKEND_PORT=5567 \
+    HADOOP_VERSION=2.7 \
+    JUPYTER_ENABLE_LAB=true \
+    SPARK_VERSION=$SPARK_VERSION \
+    SPARK_HOME=/spark \
+    PATH="${SPARK_HOME}/bin:${DOTNET_WORKER_DIR}:${PATH}"
 
 USER root
 
 COPY bin/* /usr/local/bin/
-
 COPY *.ipynb ${HOME}/dotnet.spark/examples/
-
-RUN cd / \
-    && wget -q --show-progress --progress=bar:force:noscroll https://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}.tgz \
+COPY HelloSpark /dotnet/HelloSpark
+
+RUN cd /dotnet/HelloSpark \
+    && dotnet build \
+    && cp /dotnet/HelloSpark/bin/Debug/netcoreapp${DOTNET_CORE_VERSION}/microsoft-spark-*.jar ${HOME}/ \
+    && rm -rf /dotnet/HelloSpark \
```

Comment:
The unfortunate consequence of this pattern is that HelloSpark remains in the image as a result of obtaining it via COPY.  This is not desirable.  Is there a way this can be generated during the Docker build or can it be a published tarball so that is can get copied and deleted within a single Dockerfile instruction?

Reply:
Thanks again @MichaelSimons for your great feedback! Just creating a dummy project during the build process now.